include/odp/api/spec/packet.h

```diff
@@ -2290,6 +2291,59 @@ int odp_packet_tx_compl_request(odp_packet_t pkt, const odp_packet_tx_compl_opt_
  */
 int odp_packet_has_tx_compl_request(odp_packet_t pkt);
 
+/**
+ * Packet proto stats options.
+ */
+typedef struct odp_packet_proto_stats_opt_t {
+	/** Packet proto stats object handle
+	 *
+	 * Stats in the packet proto stats object will be updated.
+	 */
+	odp_proto_stats_t stat;
+
+	/** Octet counter 0 adjust */
+	int oct_count0_adj;
+
+	/** Octet counter 1 adjust */
+	int oct_count1_adj;
+} odp_packet_proto_stats_opt_t;
+
+/**
+ * Request packet proto stats.
+ *
+ * The statistics enabled in the proto stats object are updated for the packet in
+ * packet output when the packet is transmitted or dropped. The statistics update
+ * is done as the last step of output processing after possible packet
+ * transformations (e.g. fragmentation, IPsec) that may occur. If a packet is
+ * fragmented or segmented to multiple packets as part of output processing, all
+ * the resulting packets inherit the proto stats object association from the
+ * original packet.
+ *
+ * if (sent)
+ *    counter[ODP_PROTO_STATS_ID_TX_OCT_COUNTx] += `odp_packet_len() + adjust`
+ * else if (drop)
+ *    counter[ODP_PROTO_STATS_ID_TX_OCT_COUNTx_DROP] += `odp_packet_len() + adjust`
```

Comment:
I think odp_packet_len() here is misleading since one can understand it to mean odp_packet_len(pkt) at the time of the call.

Maybe the pseudocode could be replaces with a text like this:

"The relevant octet counts will be updated with the actual packet lenght at the time of transmission or drop plus the respective adjustment value passed in the 'opt' parameter. The octet counts thus include possible additional headers added by ODP during packet output (e.g. ESP headr added by inline outbound IPsec processing). Ethernet padding and FCS are not included in the octet counts. The adjustment value is added only if the respective capability field is true and otherwise ignored."